src/components/Button/types.ts

```diff
@@ -0,0 +1,14 @@
+import React, { ReactNode } from 'react';
+import { FlattenSimpleInterpolation } from 'styled-components';
+
+export interface ButtonProps {
```

Comment:
Extending `React.HTMLProps<HTMLButtonElement>` could improve compatibility.

Reply:
I don't quite understand how it is improving compatibility as it's showing error as below. Does this error showing because HTMLButtonElement already has property 'size' as number type, not as string like in my buttonProps? Would renaming property solve this error?
<img width="702" alt="image" src="https://user-images.githubusercontent.com/55012143/181269711-bea0009c-96a8-4727-99a2-e7a826e13447.png">

Reply:
<img width="620" alt="image" src="https://user-images.githubusercontent.com/33550065/181867065-322ff694-d29c-47f2-8b60-223f3a0f6263.png">

Use `ButtonHTMLAttributes<HTMLButtonElement>` instead.